### ISSUE TITLE
Determining GIF optimization

### ISSUE THREAD

**Issue by lehitoskin** [cbowdon/gif-image#3]
There doesn't seem to be a way to determine whether a GIF has been optimized such that the first frame is used as a backdrop and the subsequent frames' data has been stripped away (and replaced with transparency) until only the parts that are different from the backdrop are kept. Some gifs appear just fine when using gif-image, yet so many more are optimized like this and I can't tell when to use the first frame as a backdrop or replace each frame with the next.

tl;dr Feature request - determine whether a GIF has been optimized

**Comment by cbowdon** [comment]
Solved by @lehitoskin 's PR. Thank you!
